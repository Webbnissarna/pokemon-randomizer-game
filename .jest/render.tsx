import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
