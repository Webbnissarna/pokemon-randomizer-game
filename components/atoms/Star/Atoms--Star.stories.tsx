import { Flex } from "theme-ui";
import Star from ".";

interface ControlProps {
  size: string;
  visible: boolean;
}

export function Default({ size, visible }: ControlProps) {
  return (
    <Flex sx={{ flexDirection: "column" }}>
      <h1>Custom</h1>
      <Star size={size} visible={visible} />
    </Flex>
  );
}

Default.args = {
  size: "150px",
  visible: true,
} as ControlProps;
