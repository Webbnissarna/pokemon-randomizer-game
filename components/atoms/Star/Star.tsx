import { Box } from "theme-ui";
import { useTheme } from "../../../utils/theme";
import { keyframes } from "@emotion/react";

interface StarProps {
  size: number | string;
  visible: boolean;
}

const showAnimation = keyframes({
  from: {
    opacity: 0,
    transform: "rotate(-90deg) scale(0.5)",
  },
  to: {
    opacity: 1,
    transform: "rotate(0deg) scale(1)",
  },
});

const loopAnimation = keyframes({
  from: {
    transform: "rotate(0deg)",
  },
  to: {
    transform: "rotate(360deg)",
  },
});

const starShapeRatio = 25.163 / 26.458; // Based on Inkscape shape

export default function Star({ size, visible }: StarProps): JSX.Element {
  const { theme } = useTheme();

  return (
    <Box
      title="star"
      sx={{
        width: size,
        willChange: "transform",
        shapeRendering: "crispEdges",

        ...(!visible && {
          transform: "scale(0)",
        }),
      }}
    >
      <Box
        sx={{
          willChange: "transform",
          shapeRendering: "crispEdges",
          ...(visible && {
            animation: `${showAnimation} 1s linear, ${loopAnimation} 10s linear 1s infinite`,
          }),
        }}
      >
        <svg
          width="100%"
          height={`${100 * starShapeRatio}%`}
          version="1.1"
          viewBox="0 0 26.458 25.163"
          xmlns="http://www.w3.org/2000/svg"
          style={{ willChange: "transform", shapeRendering: "crispEdges" }}
        >
          <path
            transform="matrix(-.27823 0 0 -.27823 1.4781 23.544)"
            d="m-42.235 84.622-11.167-34.625-36.381 0.07952 29.48-21.32-11.318-34.576 29.386 21.449 29.386-21.449-11.318 34.576 29.48 21.32-36.381-0.079517z"
            fill={theme.colors.star}
            stopColor="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2.5159"
          />
        </svg>
      </Box>
    </Box>
  );
}
