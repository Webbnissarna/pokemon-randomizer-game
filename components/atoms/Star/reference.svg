<?xml version="1.0" encoding="UTF-8"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->
<svg width="26.458mm" height="25.163mm" version="1.1" viewBox="0 0 26.458 25.163" xmlns="http://www.w3.org/2000/svg">
 <path transform="matrix(-.27823 0 0 -.27823 1.4781 23.544)" d="m-42.235 84.622-11.167-34.625-36.381 0.07952 29.48-21.32-11.318-34.576 29.386 21.449 29.386-21.449-11.318 34.576 29.48 21.32-36.381-0.079517z" fill="#fcff79" stop-color="#000000" stroke-linecap="round" stroke-linejoin="round" stroke-width="2.5159"/>
</svg>
