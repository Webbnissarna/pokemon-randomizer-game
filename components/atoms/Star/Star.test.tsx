import { render } from "../../../.jest/render";
import Star from ".";

describe("Star", () => {
  it("renders", () => {
    const { asFragment } = render(<Star size={100} visible={true} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
