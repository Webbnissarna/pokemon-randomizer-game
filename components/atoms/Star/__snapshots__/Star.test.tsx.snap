// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-19lui7g-Box"
    title="star"
  >
    <div
      class="css-xeos2n-Box"
    >
      <svg
        height="95.10545014740343%"
        style="will-change: transform; shape-rendering: crispEdges;"
        version="1.1"
        viewBox="0 0 26.458 25.163"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m-42.235 84.622-11.167-34.625-36.381 0.07952 29.48-21.32-11.318-34.576 29.386 21.449 29.386-21.449-11.318 34.576 29.48 21.32-36.381-0.079517z"
          fill="var(--theme-ui-colors-star)"
          stop-color="#000000"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2.5159"
          transform="matrix(-.27823 0 0 -.27823 1.4781 23.544)"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
